docs/design/module_structure.md

```diff
@@ -1,18 +1,25 @@
 # Module structure
 
+## Overview
 Module is the core workers in Satellite. Module is constituted by the specific extension plugins.
-There are four modules in Satellite, which is ClientManager, Gatherer, Sender, and Processor.
+There are 3 modules in one agent, which are Gatherer, Processor, and Sender.
 
-Responsibilities:
+- The Gatherer module is responsible for fetching or receiving data and pushing the data to Queue. So there are 2 kinds of Gatherer, which are ReceiverGatherer and FetcherGatherer.
+- The Processor module is responsible for reading data from the queue and processing data by a series of filter chains.
+- The Sender module is responsible for async processing and forwarding the data to the external services in the batch mode. After sending success, Sender would also acknowledge the offset of Queue in Gatherer.
 
-- ClientManager: Maintain connection and monitor connection status
-- Sender: Sender data to the external services, such as Kafka and OAP
-- Gatherer: Gather the APM data from the other systems, such as fetch prometheus metrics.
-- Processor: Data processing to create new metrics data.
+```
+                            Agent
```

Comment:
I think this should be `namespace`. Don't put 2 names in one concept.

Reply:
Agent is a whole concept:   Each namespace has one agent to collect the telemetry data. (module_design.md)

Reply:
would fix it by using one concept